eclipse/src/saros/Saros.java

```diff
@@ -200,6 +212,67 @@ public static boolean useHtmlGui() {
     return Platform.getBundle("saros.ui") != null && Boolean.getBoolean("saros.swtbrowser");
   }
 
+  /** Stops the Saros Eclipse life cycle. */
+  private void stopLifeCycle() {
+
+    final AtomicBoolean isLifeCycleStopped = new AtomicBoolean(false);
+    final AtomicBoolean isTimeout = new AtomicBoolean(false);
+
+    final Display display = Display.getCurrent();
+
+    final Thread shutdownThread =
+        ThreadUtils.runSafeAsync(
+            "dpp-shutdown", //$NON-NLS-1$
+            log,
+            () -> {
+              try {
+                lifecycle.stop();
+              } finally {
+                isLifeCycleStopped.set(true);
+
+                if (display != null) {
+                  try {
+                    display.wake();
+                  } catch (SWTException ignore) {
+                    // NOP
+                  }
+                }
+              }
+            });
+
+    isLifecycleStarted = false;
+
+    int threadTimeout = 10000;
+
+    if (display != null && !display.isDisposed()) {
+
+      display.timerExec(
+          threadTimeout,
+          () -> {
+            isTimeout.set(true);
+            display.wake();
+          });
+
+      while (isLifeCycleStopped.get() == false && isTimeout.get() == false) {
```

Comment:
`AtomicBoolean.get()` returns a `boolean`, you don't have to compare against `false`. Or was this done to improve readability?
```suggestion
      while (!isLifeCycleStopped.get() && !isTimeout.get()) {
```